### ISSUE TITLE
### TASK-01: EDA & Xử lý giá trị 0 vô lý

### ISSUE THREAD

**Issue by Minh08012005** [Minh08012005/Heart-Disease-Prediction-ML-Group-6#1]
**Assignee:** @Minh  
**Deadline:** 26/4/2026 (3 ngày)  
**Priority:** CRITICAL 🔴  
**Labels:** preprocessing, data-cleaning, eda  
**Status:** To Do  

**🎯 MỤC TIÊU CẤP TỐC:**
Cung cấp dataset đã làm sạch CHO TEAM CODE THUẬT TOÁN NGAY.

**📝 CÔNG VIỆC CỤ THỂ (Làm NGAY hôm nay):**

1. **Tải dataset** (nếu chưa có):
   - Download heart.csv từ Kaggle
   - Lưu vào `data/heart.csv`

2. **EDA nhanh (2 giờ):**
   ```python
   # notebooks/01_EDA_quick.ipynb
   import pandas as pd
   df = pd.read_csv('data/heart.csv')
   print("Shape:", df.shape)
   print("Columns:", df.columns.tolist())
   print("HeartDisease distribution:", df['HeartDisease'].value_counts())
   
   # QUAN TRỌNG: Tìm giá trị 0 vô lý
   print("Cholesterol=0:", (df['Cholesterol']==0).sum())
   print("RestingBP=0:", (df['RestingBP']==0).sum())

3. Xử lý giá trị 0 VÔ LÝ (2 giờ):__

- Cholesterol=0 → thay bằng median của NHÓM TƯƠNG ỨNG (có bệnh/không bệnh)
- RestingBP=0 → thay bằng median chung
- Lưu thành `data/heart_no_zeros.csv`
4. Tạo utils.py CƠ BẢN (1 giờ):
# src/utils.py
import numpy as np

def train_test_split(X, y, test_size=0.2, random_state=42):
    # Implement đơn giản
    pass
    
def accuracy_score(y_true, y_pred):
    return np.mean(y_true == y_pred)


__✅ DELIVERABLES (Phải có trước 26/4):__

- `data/heart.csv` (dataset gốc)
- `data/heart_no_zeros.csv` (đã xử lý giá trị 0)
- `src/utils.py` (hàm cơ bản)
- Short report: "Dataset ready for algorithm development"

__🔗 LINK TÀI LIỆU:__

- [Kaggle: Heart Failure Prediction Dataset](https://www.kaggle.com/datasets/fedesoriano/heart-failure-prediction)
- [Code mẫu xử lý giá trị 0](https://gist.github.com/...)
